pkg/kubelet/kubelet.go

```diff
@@ -1159,6 +1192,7 @@ func (kl *Kubelet) SyncPods(allPods []*api.Pod, podSyncTypes map[types.UID]metri
 	//      though the pod was not considered terminated by the apiserver).
 	// These two conditions could be alleviated by checkpointing kubelet.
 	pods := kl.filterOutTerminatedPods(allPods)
+	pods = kl.filterOutPodsPastActiveDeadline(pods)
```

Comment:
Uh...what I meant was that you only need to set the pod status to fail, and filterOutTerminatedPods would just handle the filtering. No need to duplicate code in your own function.